python/cudf/cudf/_libxx/column.pyx

```diff
@@ -51,19 +52,38 @@ cdef class Column:
             raise TypeError("Expected a Buffer for data, got " + type(data).__name__)
         if mask is not None and not isinstance(mask, Buffer):
             raise TypeError("Expected a Buffer for mask, got " + type(mask).__name__)
-        self.data = data
+        self._data = data
         self.size = size
         self.dtype = dtype
-        self.mask = mask
+        self._mask = mask
         self.offset = offset
         self.children = children
 
     @property
+    def data(self):
+        return self._data
+
+    @data.setter
+    def data(self, value):
+        self._data = value
+
+    @property
+    def mask(self):
+        return self._mask
+
+    @mask.setter
+    def mask(self, value):
+        self._nvstrings = None
```

Comment:
This sets an `_nvstrings` attribute on a column that doesn't previously have the attribute. Change this function to a one liner of `self._mask_set(value)` and then when needed in subclasses we can override / extend `def _mask_set(...)`

Reply:
Done